packages/module/src/ColumnManagementModal/ColumnManagementModal.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import {
+  Modal,
+  Button,
+  TextContent,
+  Text,
+  TextVariants,
+  DataListItem,
+  DataList,
+  DataListItemRow,
+  DataListCheck,
+  DataListCell,
+  DataListItemCells,
+  Split,
+  SplitItem,
+  ModalProps
+} from '@patternfly/react-core';
+
+export interface ColumnManagementModalColumn {
+  /** Internal identifier of a column by which table displayed columns are filtered. */
+  key: string,
+  /** The actual display name of the column possibly with a tooltip or icon. */
+  title: React.ReactNode,
+  /** If user changes checkboxes, the component will send back column array with this property altered. */
+  isShown?: boolean,
+  /** Set to false if the column should be hidden initially */
+  isShownByDefault: boolean,
+  /** The checkbox will be checked and disabled, this is applicable to identifier columns which user shouldn't be able to hide */
+  isAlwaysShown?: boolean
+}
+
+export interface ColumnManagementModalProps extends Omit<ModalProps, 'ref' | 'children'> {
+  /** Flag to show the modal */
+  isModalOpen: boolean,
+  /** Invoked when modal visibility is changed */
+  setModalOpen: (shouldOpen: boolean) => void
+  /** Current column state */
+  appliedColumns: ColumnManagementModalColumn[],
+  /** Invoked with new column state after save button is clicked */
+  applyColumns: (newColumns: ColumnManagementModalColumn[]) => void,
+};
+
+const ColumnManagementModal: React.FunctionComponent<ColumnManagementModalProps> = (
+  { isModalOpen,
+    setModalOpen,
+    appliedColumns,
+    applyColumns,
+    ...props }: ColumnManagementModalProps) => {
```

Comment:
Can you please add a default value of "ColumnManagementModal" to `ouiaId` property and also ouiaIds to important sub-elements (checkboxes, buttons)? We've just made some updates in OUIA ids across the components - see https://github.com/patternfly/react-component-groups?tab=readme-ov-file#testing and #131 
Thank you :-)